### DISCUSSION TITLE
Very large repositories may not be indexed. What is considered a very large repository?

### DISCUSSION THREAD

**Discussion by carlosdoordash** [community/community#64775] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Question

### Body

[GitHub Code Search Limitations](https://docs.github.com/en/search-github/github-code-search/about-github-code-search#limitations) says the following...

> The current limitations on indexed code are:
> 
> - Vendored and generated code is excluded
> - Empty files and files over 350 KiB are excluded
> - Lines over 1,024 characters long are truncated
> - Only UTF-8 encoded files are included
> - Very large repositories may not be indexed

Can you elaborate on what would be considered a very large repository? Is there a max size of a .git directory, or a max number of files in the repository, or some other limit?

**Comment by look** [comment, marked as answer]
The exact limits are subject to change which is why we haven't documented them, but it looks at both the size of the Git repository and the number of paths. The size limit is around 50 GB and the path limit varies by repo but is at least 75,000.

**Reply by mesheets** [reply]
If getting a “[repository] cannot be searched because it is too large” message (like the following),
<img width="495" height="25" alt="image" src="https://github.com/user-attachments/assets/f7ea4b2b-f70b-4729-81e2-191823fcca0b" />

…how can maintainers determine what exactly about the repository is considered “too large”?

The repository shown in the screenshot above is reported as being just over 200 MB, so that should be well under the noted 50 GB threshold:
<img width="295" height="208" alt="image" src="https://github.com/user-attachments/assets/e1ac317f-5dd7-43c8-bd28-a7f922e1df06" />

As it stands, there is no indication as what maintainers might need to do to ensure that a repository like this can be indexed.

There have been no issues with MAX_PATH path character length on Windows, so that path limit shouldn't be an issue.

Folder count is 5,600+ folders and file count is 109,000+ files, but given how this is all part of a standard, combined GNU toolchain build, there is not much that can be done to reduce those counts (if file counts were part of the issue).

Thank you!
